src/amperity/gocd/secret/vault/plugin.clj

```diff
@@ -171,6 +175,7 @@
       "aws-iam"
       (vault/authenticate! client :aws-iam
                            {:iam-role    (:iam_role inputs)
+                            :sts-region  (:sts_region inputs)
```

Comment:
Based on your change to the client library, you need to not set this if it's blank instead of always passing it.

Reply:
FWIW, when unset in the UI, `(:sts_region inputs)` is `nil` and does flow through to `vault-clj-aws` correctly (i.e. auth succeeds/fails as expected based on what is present in this field and the configuration of the auth backed in the configured Vault API).

Reply:
Have, however, taken your suggestion at https://github.com/amperity/vault-clj-aws/pull/2#pullrequestreview-418578870.